### ISSUE TITLE
Add OSX support 

### ISSUE THREAD

**Issue by ericdorsey** [ericdorsey/HostMAC#9]
Might as well keep up the "all platforms supported" trend. Currently doesn't work for OSX, doesn't detect IP and falls to the default 127.0.0.1 from `detect_ip()`:

```
~/Code/HostMAC (master) $ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.10.2
BuildVersion:   14C1510
~/Code/HostMAC (master) $ python hostmac.py

Detected IP: 127.0.0.1

1) Continue with detected IP (creates 254 entries)
2) Enter another IP (creates one entry)
3) Exit

Selection? 3
```

**Comment by ericdorsey** [comment]
Copied part of convo from: https://github.com/ericdorsey/HostMAC/pull/13#issuecomment-83230361

for OSX, we have to adjust the `arp` request it looks like. Works a bit diff than Ubuntu:

```
# OSX 

~ $ arp -a 192.168.1.1
usage: arp [-n] [-i interface] hostname
       arp [-n] [-i interface] [-l] -a
       arp -d hostname [pub] [ifscope interface]
       arp -d [-i interface] -a
       arp -s hostname ether_addr [temp] [reject] [blackhole] [pub [only]] [ifscope interface]
       arp -S hostname ether_addr [temp] [reject] [blackhole] [pub [only]] [ifscope interface]
       arp -f filename
```

We can get it with 

```
# also OSX 

~ $ arp -a | grep 192.168.1.1
? (192.168.1.1) at ff:ee:ff:aa:00:00 on en0 ifscope [ethernet]
```

But Ubuntu returns a bunch of matches with that:

```
# Ubuntu 

$ arp -a | grep 192.168.1.1
? (192.168.1.12) at <incomplete> on eth0
? (192.168.1.1) at ff:ee:ff:aa:00:00 [ether] on eth0
? (192.168.1.15) at <incomplete> on eth0
? (192.168.1.14) at <incomplete> on eth0
```

But I think this will work on both OSX and *nix

```
# returns just one line on Ubuntu and OSX

$ arp -a | grep -w 192.168.1.1
```

So thinking we want to update `getMac()` to use `arp -a | grep -w {current IP}` for *nix / posix.

**Comment by ericdorsey** [comment]
Merged in c4ae523

When run from OSX:
- Uses native OSX `smbutil` to try to get a NetBIOS name
- Adds a fallback `try / except` block utilizing `s.connect(('8.8.8.8', 80))` to get IP of OSX host (without user having to manually enter IP for OSX)
- More robust `nslooky(ip)`
  - would like to run some more tests in an environment that has DNS setup, see if hostnames come back properly from *nix and OSX hosts with the `socket.gethostbyaddr(ip)` call

**Comment by bcambl** [comment]
Do you still see value in keeping:

```
    try:
        s.connect(('1.2.3.4', 0))
        ip_address = s.getsockname()[0]
```

**Comment by ericdorsey** [comment]
Good question; I was wondering if certain OS's don't like connecting on certain ports. I left both in with my last merge because it tested good on Windows and OSX with the fallback `try / catch`, but didn't get too far into the what prevented OSX from working with the initial `s.connect(('1.2.3.4', 0))` config. 

The resolution ironically ended up taking me to using a legit target (the Google DNS IP) and port (80) for the OSX fix. 

Did you by chance test Windows with `s.connect(('8.8.8.8', 80))` only? I havn't yet :)

**Comment by ericdorsey** [comment]
Closed with 8a6a7e8560b0837de75d2a190d2edd40b09fecfe
